### ISSUE TITLE
[Feature request] Add FileContent Configuration

### ISSUE THREAD

**Issue by Kahbazi** [aspnet/Extensions#1086]
Is this a good feature to have a configuration source like KeyPerFile which reads content of a file but it's only for one file and the key can be set via code?

This is what I have in mind

`public static IConfigurationBuilder AddFileContent(this IConfigurationBuilder builder, string filePath, string key, bool optional, bool reloadOnChange)
`
I know the implementation is simple, but is this something that could be added in Microsoft.Extensions?

**Comment by anurse** [comment]
I don't think this is something we're looking for in the core config providers. It seems reasonable to have as a third-party extension, but not something we've heard a lot of demand for in the core library.
